File: api-posts/src/main/java/com/github/alexeysol/geekregime/apiposts/configurations/IndexerConfiguration.java
package com.github.alexeysol.geekregime.apiposts.configurations;

import com.github.alexeysol.geekregime.apiposts.utils.search.Indexer;
import org.springframework.boot.ApplicationArguments;
import org.springframework.boot.ApplicationRunner;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class IndexerConfiguration {
    @Bean
    public ApplicationRunner buildIndex(Indexer indexer) {
        return (ApplicationArguments args) ->
            indexer.indexPersistedData("com.github.alexeysol.geekregime.apiposts.models.entities.Post");
    }
}

File: api-users/src/main/kotlin/com/github/alexeysol/geekregime/apiusers/configurations/IndexerConfiguration.kt
package com.github.alexeysol.geekregime.apiusers.configurations

import com.github.alexeysol.geekregime.apiusers.utils.search.Indexer
import org.springframework.boot.ApplicationRunner
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class IndexerConfiguration {
    @Bean
    fun buildIndex(indexer: Indexer): ApplicationRunner {
        return ApplicationRunner {
            indexer.indexPersistedData("com.github.alexeysol.geekregime.apiusers.models.entities.User")
        }
    }
}
